cylc/flow/job_pool.py

```diff
@@ -135,22 +132,18 @@ def remove_job(self, job_d):
         point, name, sub_num = self.parse_job_item(job_d)
         t_id = f'{self.workflow_id}{ID_DELIM}{point}{ID_DELIM}{name}'
         j_id = f'{t_id}{ID_DELIM}{sub_num}'
-        try:
+        if j_id in self.pool or j_id in self.updates:
             self.task_jobs[t_id].discard(j_id)
             self.deltas.pruned.append(j_id)
             self.updates_pending = True
-        except KeyError:
-            pass
```

Comment:
@dwsutherland it seems good to not catch TypeError and AttributeError in the other try blocks, because those indicate unexpected errors that should not happen (is that right?) but these ones (KeyError) are fine as-is because they're simply a short-cut way to distinguish keys that may not (correctly) be present in the data structures.  In fact doing this with a try block is generally considered better than using an `if` test (not sure if there's much of a performance advantage, but might be some).

Reply:
Yes, right.. Perhaps I should try without checking.. I think it was there for reloads/restarts, we'll see.

Reply:
Fixed.. Yes needed for job restart.